src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyStageFileItem.scala

```diff
@@ -69,69 +70,101 @@ object EasyStageFileItem {
     }
 
   def run(implicit s: FileItemSettings): Try[Unit] = {
-    log.debug(s"executing: $s")
+    trace(s)
     for {
       datasetId        <- getValidDatasetId(s)
       sdoSetDir        <- mkdirSafe(s.sdoSetDir)
       datasetSdoSetDir <- mkdirSafe(new File(sdoSetDir, datasetId.replace(":", "_")))
-      (parentId, parentPath, newElements)  <- getPathElements()
-      items            <- Try { getItemsToStage(newElements, datasetSdoSetDir, parentId) }
-      _                = log.debug(s"Items to stage: $items")
-      _                <- Try{items.init.foreach { case (sdo, path, parentRelation) => createFolderSdo(sdo, relPath(parentPath, path), parentRelation) }}
-      _                <- items.last match {case (sdo, _, parentRelation) => createFileSdo(sdo, parentRelation) }
+      existingAncestor <- s.easyFilesAndFolders.getExistingAncestor(s.pathInDataset.get, datasetId)
+      _                = createFolderSdos(existingAncestor, s.pathInDataset.get, datasetSdoSetDir)
+      _                <- createFileSdoForExistingDataset(datasetSdoSetDir, existingAncestor)
     } yield ()
   }
 
-  def relPath(parentPath: String, path: String): String =
-    if (parentPath.isEmpty) new File(path).toString // prevent a leading slash
-    else new File(parentPath, path).toString
-
-  def getPathElements()(implicit s: FileItemSettings): Try[(String, String, Seq[String])] = {
-    val file = s.pathInDataset.get
-    s.easyFilesAndFolders.getExistingAncestor(file, s.datasetId.get)
-      .map { case (parentPath, parentId) =>
-        log.debug(s"Parent in repository: $parentId $parentPath")
-        val newItems = file.toString.replaceFirst(s"^$parentPath/", "").split("/")
-        (parentId, parentPath, newItems.toSeq)
-      }
-  }
-
-  def getItemsToStage(pathElements: Seq[String], datasetSdoSet: File, existingFolderId: String): Seq[(File, String, (String, String))] = {
+  def getItemsToStage(pathElements: Seq[String],
+                      datasetSdoSet: File,
+                      existingFolderId: String
+                     ): Seq[(File, String, RelationObject)] = {
+    trace(pathElements, datasetSdoSet, existingFolderId)
     getPaths(pathElements)
-    .foldLeft(Seq[(File, String, (String, String))]())((items, path) => {
+    .foldLeft(Seq[(File, String, RelationObject)]())((items, path) => {
       items match {
-        case s@Seq() => s :+ (new File(datasetSdoSet, toSdoName(path)), path, "object" -> s"info:fedora/$existingFolderId")
+        case s@Seq() =>
```

Comment:
these two cases look very much alike (apart from the mapping in the third lines)